### PR TITLE
document what `declare` does

### PR DESCRIPTION
**Pull request by evmar**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,6 +29,30 @@ well.
...
+`declare ...`, is intepreted by Tsickle as a name that should be preserved
+through Closure compilation (i.e. not renamed into something shorter).  Use it
+any time the specific string names of your fields are important.
+
+Example:
+
+    declare interface JSONResult {
+        username: string;
+    }
+    let r = JSON.parse(input) as JSONResult;
```
Comment (rkirov) [inline README.md]: The example doesn't show why username is "important". Add a r.username access.

File: README.md [hunk 2]
```diff
@@ -29,6 +29,30 @@ well.
...
+
+Closure and TypeScript are not identical.  Tsickle hides most of the
+differences, but users must still be aware of some differences.
+
+#### `declare`
+
+Any declaration in a `.d.ts` file, as well as any declaration tagged with
+`declare ...`, is intepreted by Tsickle as a name that should be preserved
+through Closure compilation (i.e. not renamed into something shorter).  Use it
+any time the specific string names of your fields are important.
```
Comment (rkirov) [inline README.md]: "important" is too vague for my liking, but I understand the desire to capture the behavior in a single sentence. How about "names of the fields are meaningful to something outside the TypeScript code (third-party library, backend, etc)."

File: README.md (hunk 1 above)
Comment (rkirov) [inline README.md]: There is also a co-example when the object is constructed in the TypeScript code but passed out of the compilation scope.

```
let r = {username: ...};
window.fetch(..., r);
```

Comment (evmar) [inline README.md]: Done.

File: README.md (hunk 2 above)
Comment (evmar) [inline README.md]: Done.

**Comment by rkirov** [comment]
LGTM

**Pull request merged**